homeassistant/components/edl21/sensor.py

```diff
@@ -410,6 +411,14 @@ class EDL21Entity(SensorEntity):
 
     _attr_should_poll = False
 
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._electricity_id)},
+            name=self._electricity_id,
+        )
```

Comment:
This is information that doesn't changes after initialization.
You should move to to __init__ method and replace the property with the attribute;
`self._attr_device_info = DeviceInfo(...)`

Maybe also take the opportunity to migrate `name` and `unique_id` properties to use attributes instead.